### PR TITLE
feat(@angular/cli): Update command runner.

### PR DESCRIPTION
**Pull request by Brocco**
This change removes the dependency of the ember-cli command structure.

### CODE REVIEW

File: packages/@angular/cli/commands/easter-egg.ts [hunk 1]
```diff
@@ -1,35 +1,28 @@
...
     const phrase = pickOne([
       `You're on it, there's nothing for me to do!`,
       `Let's take a look... nope, it's all good!`,
       `You're doing fine.`,
       `You're already doing great.`,
       `Nothing to do; already awesome. Exiting.`,
-      `Error 418: As Awesome As Can Get.`
+      `Error 418: As Awesome As Can Get.`,
+      `I spy with my little eye a great developer!`,
+      `noop... already awesome.`
```
Comment (hansl) [inline packages/@angular/cli/commands/easter-egg.ts]: Capital letter?

File: packages/@angular/cli/ember-cli/lib/cli/cli.js [hunk 2]
```diff
@@ -76,169 +83,61 @@ class CLI {
...
+    const logger = new logging.IndentLogger('cling');
+    let loggingSubscription;
+    if (!this.testing) {
+      loggingSubscription = this.initializeLogging(logger);
+    }
+    const context = {
+      project: environment.project,
+      ui: this.ui,
+    };
+    return runCommand(environment.commands, environment.cliArgs, logger, context)
```
Comment (hansl) [inline packages/@angular/cli/ember-cli/lib/cli/cli.js]: Are we doing `await`/`async` in this PR?

File: packages/@angular/cli/ember-cli/lib/cli/lookup-command.js [hunk 3]
```diff
@@ -24,6 +24,8 @@ module.exports = function(commands, commandName, commandArgs, optionHash) {
   // Attempt to find command in ember-cli core commands
   let command = findCommand(commands, commandName);
 
+
```
Comment (hansl) [inline packages/@angular/cli/ember-cli/lib/cli/lookup-command.js]: Remove empty new lines.

File: packages/@angular/cli/models/command.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      this.ui = context.ui;
+    }
+  }
+
+  initialize(_options: any): Promise<void> {
+    return Promise.resolve();
+  }
+
+  validate(_options: any): boolean | Promise<boolean> {
+    return true;
```
Comment (hansl) [inline packages/@angular/cli/models/command.ts]: No need for this, right?

File: packages/@angular/cli/commands/get.ts [hunk 5]
```diff
@@ -1,49 +1,47 @@
...
 
-
-const GetCommand = Command.extend({
-  name: 'get',
-  description: 'Get a value from the configuration. Example: ng get [key]',
-  works: 'everywhere',
-
-  availableOptions: [
+export default class GetCommand extends Command {
+  public name = 'get';
```
Comment (hansl) [inline packages/@angular/cli/commands/get.ts]: readonly?

File: packages/@angular/cli/ember-cli/lib/cli/cli.js (hunk 2 above)
Comment (Brocco) [reply packages/@angular/cli/ember-cli/lib/cli/cli.js]: Yes, I can refactor, but I wanted it working first.

File: packages/@angular/cli/models/command.ts (hunk 4 above)
Comment (Brocco) [reply packages/@angular/cli/models/command.ts]: It's a base implementation that can be overridden, but the runner relies on a boolean being returned. Not all commands will implement validate.

File: packages/@angular/cli/commands/get.ts (hunk 5 above)
Comment (Brocco) [reply packages/@angular/cli/commands/get.ts]: Won't hurt anything by adding it.

**Comment by Brocco** [comment]
When this PR gets merged, this devkit PR should be merged too: angular/devkit#481

File: packages/@angular/cli/models/command.ts (hunk 4 above)
Comment (hansl) [reply packages/@angular/cli/models/command.ts]: Sorry I pointed to the wrong file. the Doc command still has an empty override.

**Pull request merged**